charmcraft/manifest.py

```diff
@@ -0,0 +1,80 @@
+# Copyright 2020-2021 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For further info, check https://github.com/canonical/charmcraft
+
+"""Charmcraft manifest.yaml related functionality."""
+
+import datetime
+import logging
+import pathlib
+from typing import Optional
+
+import yaml
+
+from charmcraft import __version__, config, utils
+from charmcraft.cmdbase import CommandError
+
+logger = logging.getLogger(__name__)
+
+
+def create_manifest(
+    basedir: pathlib.Path,
+    started_at: datetime.datetime,
+    bases_config: Optional[config.BasesConfiguration] = None,
+):
+    """Create manifest.yaml in basedir for given base configuration.
+
+    :param basedir: Directory to create Charm in.
+    :param started_at: Build start time.
+    :param bases_config: Relevant bases configuration.
+
+    :returns: Path to created manifest.yaml.
+    """
+    if bases_config is None:
+        os_platform = utils.get_os_platform()
+
+        # XXX Facundo 2021-03-29: the architectures list will be provided by the caller when
+        # we integrate lifecycle lib in future branches
+        architectures = [utils.get_host_architecture()]
+
+        # XXX Facundo 2021-04-19: these are temporary translations until charmcraft
+        # changes to be a "classic" snap
```

Comment:
Maybe this is an opportunity to clean this? we're not *strict* anymore, right?

Reply:
I was avoiding unnecessary refactoring in the migration, but may as well rip that bandaid...

Done!